lib/MySQL_Session.cpp

```diff
@@ -6491,10 +6499,13 @@ bool MySQL_Session::handler___status_WAITING_CLIENT_DATA___STATE_SLEEP___MYSQL_C
 #endif
 			if (index(dig,';') && (index(dig,';') != dig + strlen(dig)-1)) {
 				string nqn;
-				if (mysql_thread___parse_failure_logs_digest)
+				if (mysql_thread___log_mysql_statements == false) {
+					nqn = ALTERNATE_TEXT;
+				} else if (mysql_thread___parse_failure_logs_digest) {
 					nqn = string(CurrentQuery.get_digest_text());
-				else
+				} else {
 					nqn = string((char *)CurrentQuery.QueryPointer,CurrentQuery.QueryLength);
+				}
```

Comment:
I do not have any strong opinion here but I think we can use `assign` function here to assign the value to the string object. 

The `std::string` constructors used here create a temporary object and then is copied or moved to `nqn`. Using `assign()` directly modifies the existing `std::string` object, avoiding this overhead.

```
if (!mysql_thread___log_mysql_statements) {
    nqn = ALTERNATE_TEXT;
} else if (mysql_thread___parse_failure_logs_digest) {
    const char* digest_text = CurrentQuery.get_digest_text();
    nqn.assign(digest_text);
} else if (CurrentQuery.QueryPointer && CurrentQuery.QueryLength > 0) {
    nqn.assign((char *)CurrentQuery.QueryPointer, CurrentQuery.QueryLength);
}
```